core/trino-main/src/main/java/io/trino/execution/resourcegroups/InternalResourceGroup.java

```diff
@@ -733,6 +741,10 @@ private void startInBackground(ManagedQueryExecution query)
             updateEligibility();
             executor.execute(query::startWaitingForResources);
         }
+        // mark the time when the query was started for this group and all ancestors
+        for (InternalResourceGroup group = this; group != null; group = group.parent.orElse(null)) {
+            group.getStartedQueries().update(1);
```

Comment:
I think you can update this counter in the loop above (and once before it). I assume your intention was to avoid doing this while the lock is held, but since all callers of `startInBackground` hold this lock, this optimization will not work in practice.

Reply:
Great point!